service/tcp.go

```diff
@@ -62,12 +62,13 @@ func remoteIP(conn net.Conn) netip.Addr {
 	return netip.Addr{}
 }
 
-// Wrapper for logger.Debugf during TCP access key searches.
-func debugTCP(cipherID, template string, val interface{}) {
+// Wrapper for slog.Debug during TCP access key searches.
+func debugTCP(template string, cipherID string, args ...any) {
 	// This is an optimization to reduce unnecessary allocations due to an interaction
-	// between Go's inlining/escape analysis and varargs functions like logger.Debugf.
-	if logger.IsEnabledFor(logging.DEBUG) {
-		logger.Debugf("TCP(%s): "+template, cipherID, val)
+	// between Go's inlining/escape analysis and varargs functions like slog.Debug.
+	if slog.Default().Enabled(context.TODO(), slog.LevelDebug) {
```

Comment:
I don't know if this works correctly. The methods that don't take a context pass nil, not TODO. The Enabled method uses the Background context: https://cs.opensource.google/go/x/exp/+/0cdaa3ab:slog/logger.go;l=134

I think we should pass nil or the Background context.